client/components/deposits-overview/recent-deposits.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { Flex, FlexItem, Icon } from '@wordpress/components';
+import { calendar } from '@wordpress/icons';
+
+/**
+ * Internal dependencies.
+ */
+import strings from './strings';
+import './style.scss';
+import DepositStatusChip from 'components/deposit-status-chip';
+import { getDepositDate } from 'deposits/utils';
+import { CachedDeposit } from 'wcpay/types/deposits';
+import { useDeposits } from 'wcpay/data';
+import { formatCurrency } from 'wcpay/utils/currency';
+import { Link } from '@woocommerce/components';
+import { getDetailsURL } from 'wcpay/components/details-link';
+
+interface DepositRowProps {
+	deposit: CachedDeposit;
+}
+interface RecentDeposits {
+	deposits: CachedDeposit[];
+	isLoading: boolean;
+}
+interface RecentDepositsProps {
+	currency: string | undefined;
+}
+
+const tableClass = 'wcpay-deposits-overview__table';
+
+const GetRecentDeposits = ( currency?: string ): RecentDeposits => {
+	const query = {
+		status_is_not: 'estimated',
+		store_currency_is: currency,
+		orderby: 'date',
+		order: 'desc',
+		per_page: '3',
+	};
+	const deposits = useDeposits( query );
+
+	return {
+		deposits: deposits.deposits,
+		isLoading: deposits.isLoading,
+	};
+};
+
+const DepositTableRow: React.FC< DepositRowProps > = ( {
+	deposit,
+}: DepositRowProps ): JSX.Element => {
+	return (
+		<Flex className={ `${ tableClass }__row` }>
+			<FlexItem className={ `${ tableClass }__cell` }>
+				<Icon icon={ calendar } size={ 17 } />
+				<Link href={ getDetailsURL( deposit.id, 'deposits' ) }>
+					{ getDepositDate( deposit ) }
+				</Link>
+			</FlexItem>
+			<FlexItem className={ `${ tableClass }__cell` }>
+				<DepositStatusChip status={ deposit.status } isCompact />
+			</FlexItem>
+			<FlexItem className={ `${ tableClass }__cell` }>
+				{ formatCurrency( deposit.amount, deposit.currency ) }
+			</FlexItem>
+		</Flex>
+	);
+};
+
+/**
+ * Renders the Recent Deposit details component.
+ *
+ * This component includes the recent deposit heading, table and notice.
+ *
+ * @param {DepositRowProps} props Next Deposit details props.
+ * @return {JSX.Element} Rendered element with Next Deposit details.
+ */
+const RecentDeposits: React.FC< RecentDepositsProps > = ( {
+	currency,
+}: RecentDepositsProps ): JSX.Element => {
+	const recentDeposits = GetRecentDeposits( currency );
+	const isLoading = recentDeposits.isLoading;
+
+	if ( isLoading || recentDeposits.deposits.length === 0 ) {
+		return <></>;
```

Comment:
Thinking out loud: [returning `null` is a more common pattern](https://testing-library.com/docs/queries/about#types-of-queries) than returning an empty fragment.

However, I notice that `<></>` is used throughout WCPay, perhaps to stick to `JSX.Element` return type.

Reply:
> However, I notice that <></> is used throughout WCPay, perhaps to stick to JSX.Element return type.

Curious, if we continue doing this (using empty fragments instead of null) I want to understand the benefit/rationale.

Reply:
At a glance, my understanding is that this is mainly to appease TS:
* `<></>` is a valid return for the `JSX.Element` type.
* `null` is not a `JSX.Element`, we'd have to add `JSX.Element | null` to the function return type (IMO good thing, more explicit).

I think both options clearly communicate the intention (return empty/nothing).

<details>
<summary>There is a negligible performance hit to using `<></>` compared to `null`</summary>

In this contrived example, you'll see that `<ReturnFragment />` is rendered, shown in the React profiler as `0.1ms`, while `<ReturnNull />` isn't rendered.

<img width="329" alt="Screenshot 2023-04-05 at 10 23 02" src="https://user-images.githubusercontent.com/3147296/229950871-1c8bd19b-ab65-42c4-ba60-5e306079f764.png">

<img width="220" alt="image" src="https://user-images.githubusercontent.com/3147296/229950819-b3f775b4-ff1a-4c10-b230-9ca52e874b99.png">

</details>

Reply:
Gotcha – it's simpler to guarantee `JSX.Element` return type. Works for me :)